### PR TITLE
3040 timeline ux improvements

### PR DESCRIPTION
**Pull request by despo**
## Context

This change maps timeline events to the corresponding paths

## Link to Trello card

https://trello.com/c/T5T3fS4t/3040-timeline-ux-improvements

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/components/provider_interface/application_timeline_component.html.erb [hunk 1]
```diff
@@ -4,14 +4,16 @@
     <div class="app-timeline__item">
       <div class="app-timeline__header">
         <h2 class="app-timeline__title"><%= event.title %></h2>
-        <p class="app-timeline__byline">by <%= event.actor %></p>
       </div>
-
-      <p class="app-timeline__date">
-        <time datetime="<%= event.date.to_s(:govuk_date_short_month) %>"><%= event.date.to_s(:govuk_date_short_month) %></time>
+      <p class="app-timeline__actor_and_date">
+      <%= event.actor %>,
```
Comment (duncanjbrown) [inline app/components/provider_interface/application_timeline_component.html.erb:9]: Indent pls! 🙏

File: app/components/provider_interface/application_timeline_component.rb [hunk 2]
```diff
@@ -32,6 +32,7 @@ def status_change_events
           title_for(change),
           actor_for(change),
           change.changed_at,
+          link_params_for_status(change),
```
Comment (duncanjbrown) [inline app/components/provider_interface/application_timeline_component.rb:35]: I liked `link_name` and `link_path`! Appreciate they're easier to represent in arrays. Could these be splatted here maybe?

**Review by duncanjbrown — COMMENTED** [review]
Code looks good and working well — nice dovetailing with #3448!

One q about keeping everything on the surface of the struct, which I think could keep everything readable?

File: app/components/provider_interface/application_timeline_component.rb (hunk 2 above)
Comment (despo) [reply app/components/provider_interface/application_timeline_component.rb]: Addressed.

File: app/components/provider_interface/application_timeline_component.html.erb (hunk 1 above)
Comment (despo) [reply app/components/provider_interface/application_timeline_component.html.erb]: 👍

**Review by duncanjbrown — APPROVED** [review]

**Pull request merged**